Gateway/Request/CustomerIpDataBuilder.php

```diff
@@ -52,7 +53,12 @@ public function build(array $buildSubject)
     {
         /** @var \Magento\Payment\Gateway\Data\PaymentDataObject $paymentDataObject */
         $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
-        $shopperIp = $paymentDataObject->getPayment()->getOrder()->getXForwardedFor();
+        $order = $paymentDataObject->getPayment()->getOrder();
+        $shopperIp = $order->getXForwardedFor();
```

Comment:
```suggestion
        $shopperIp = $order-> getRemoteIp();
```
`getRemoteIp` actually uses the `$_SERVER['REMOTE_ADDR']` server variable, which is more reliable as a client IP address.